sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor{TPartition}.cs

```diff
@@ -0,0 +1,433 @@
+        ///   immediately.  If there were not a sufficient number of events available in the partition to populate a full batch, the event processor will continue reading from the partition
+        ///   to reach the requested batch size until the <see cref="EventProcessorOptions.MaximumWaitTime"/> has elapsed, at which point it will return a batch containing whatever events were
+        ///   available by the end of that period.
+        ///
+        ///   If a <see cref="EventProcessorOptions.MaximumWaitTime"/> was not requested, indicated by setting the option to <c>null</c>, the event processor will continue reading from the Event Hub
+        ///   partition until a full batch of the requested size could be populated and will not dispatch any partial batches to this method.</para>
+        ///
+        ///   <para>Should an exception occur within the code for this method, the event processor will allow it to bubble and will not surface to the error handler or attempt to handle
+        ///   it in any way.  Developers are strongly encouraged to take exception scenarios into account and guard against them using try/catch blocks and other means as appropriate.</para>
+        /// </remarks>
+        ///
+        protected abstract Task OnProcessingEventBatchAsync(IEnumerable<EventData> events,
+                                                            TPartition partition,
+                                                            CancellationToken cancellationToken);
+
+        /// <summary>
+        ///   Performs the tasks needed when an unexpected exception occurs within the operation of the
+        ///   event processor infrastructure.
+        /// </summary>
+        ///
+        /// <param name="exception">The exception that occurred during operation of the event processor.</param>
+        /// <param name="partition">The context of the partition associated with the error, if any; otherwise, <c>null</c>.</param>
+        /// <param name="operationDescription">A short textual description of the operation during which the exception occurred; intended to be informational only.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> instance to signal the request to cancel the processing.  This is most likely to occur when the processor is shutting down.</param>
+        ///
+        /// <remarks>
+        ///   This error handler is invoked when there is an exception observed within the event processor itself; it is not invoked for exceptions in
+        ///   code that has been implemented to process events or other overrides and extension points.  The event processor will make every effort
+        ///   to recover from exceptions and continue processing.  Should an exception that cannot be recovered from be encountered, the processor
+        ///   will forfeit ownership of all partitions that it was processing so that work may redistributed.
```

Comment:
Is it more accurate to state "will *attempt* to forfeit" ?

Reply:
Yes.  Yes, I believe that would be.